File: MojPrijevozFront/moj_prijevoz/lib/resources/requests/user/reset_password_request.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:moj_prijevoz/resources/requests/user/request_reset_password_request.dart';

part 'reset_password_request.g.dart';

@JsonSerializable()
class ResetPasswordRequest extends RequestResetPasswordRequest {
  String? code;
  String? password;
  String? passwordAgain;

  ResetPasswordRequest({
    super.email,
    this.code,
    this.password,
    this.passwordAgain,
  });

  @override
  Map<String, dynamic> toJson() => _$ResetPasswordRequestToJson(this);

  factory ResetPasswordRequest.fromJson(Map<String, dynamic> json) =>
      _$ResetPasswordRequestFromJson(json);
}

File: MojPrijevoz/MojPrijevoz.Model/Requests/User/ResetPasswordRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MojPrijevoz.Model.Requests.User;

public class ResetPasswordRequest
{
    [Required] public string Email { get; set; } = null!;

    [Required] public string Code { get; set; } = null!;

    [Required] public string Password { get; set; } = null!;

    [Required] public string PasswordAgain { get; set; } = null!;
}